templates/include/yarp/ast.h.erb

```diff
@@ -55,7 +55,7 @@ typedef enum {
 // We store the node type enum in every node in the tree. We don't have nearly
 // as many node types as we do bits in an enum, so we're going to use the top
 // half of the enum to store flags about the node.
-static const unsigned int YP_NODE_FLAGS_SHIFT = sizeof(yp_node_type_t) * 8 / 2;
```

Comment:
`sizeof(yp_node_type_t) * 8 / 2` wasn't being evaluated at compile time. Accordingly, `YP_NODE_FLAG_NEWLINE` wasn't being evaluated at compile time, making it not const. 

Changing to use a macro means `YP_NODE_FLAGS_SHIFT` will be replaced at compile time in `YP_NODE_FLAG_NEWLINE` so it'll still be a `const `

Reply:
https://github.com/ruby/yarp/pull/1178#discussion_r1285872849

Reply:
So I think YP_NODE_FLAG_NEWLINE needs to be a macro too